chronosphere/enum/promql_matcher_type.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Chronosphere Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enum
+
+import (
+	configunstable "github.com/chronosphereio/terraform-provider-chronosphere/chronosphere/pkg/configunstable/models"
+)
+
+// PromQLMatcherType is an enum.
+var PromQLMatcherType = newEnum("PromQLMatcherType", []value[configunstable.ConfigunstablePromQLMatcherType]{
+	{
+		isDefault: true,
+	},
+	{
+		v1:    configunstable.ConfigunstablePromQLMatcherTypeMatchEqual,
+		alias: "MatchEqual",
```

Comment:
do we expect to have values that aren't "match"? wonder if we can simplify to "Equal" "NotEqual" etc
this naming also differs slightly from the existing matcher type which uses EXACT
https://github.com/chronosphereio/terraform-provider-chronosphere/blob/819a3249f331d1650d358db75fec51170bef1a0f/chronosphere/enum/matcher_type.go#L30